packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -0,0 +1,83 @@
+import {IReputation, ReputationStore} from "../IReputation";
+import {BeaconBlock, BeaconBlockHeader, Checkpoint, Epoch, Slot} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {hashTreeRoot} from "@chainsafe/ssz";
+import {blockToHeader} from "../../chain/stateTransition/util";
+import {IReqResp} from "../../network";
+
+export function isValidHeaderChain(config: IBeaconConfig, start: BeaconBlockHeader, blocks: BeaconBlock[]): boolean {
+  blocks = blocks.sort((b1, b2) => b1.slot - b2.slot);
+  let previousRoot = hashTreeRoot(start, config.types.BeaconBlockHeader);
+  for(const block of blocks) {
+    if(!previousRoot.equals(block.parentRoot)) {
+      return false;
+    }
+    previousRoot = hashTreeRoot(blockToHeader(config, block), config.types.BeaconBlockHeader);
```

Comment:
this should be signingRoot, also we don't need the extra `blockToHeader`, we can signingRoot the BeaconBlock directly.